fhir-server/src/main/java/au/csiro/pathling/fhirpath/element/IntegerPath.java

```diff
@@ -157,7 +165,7 @@ public static Function<Numeric, NonLiteralPath> buildMathOperation(@Nonnull fina
           }
           return ElementPath
               .build(expression, dataset, idColumn, eidColumn, valueColumn, true, Optional.empty(),
-                  thisColumn, fhirType);
+                  thisColumn, source.getFhirType());
         case DIVISION:
           final Column numerator = source.getValueColumn().cast(DecimalPath.getDecimalType());
           valueColumn = operation.getSparkFunction().apply(numerator, targetValueColumn);
```

Comment:
Perhaps can be delegated to a counterpart of Comparator for arithmetic operations.

Reply:
Ignored (not obvious how that could be done).